src/App.tsx

```diff
@@ -2,108 +1,64 @@
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalItems = items.length;
+
+  const from = (currentPage - 1) * Number(perPage) + 1;
+  const to =
+    currentPage * Number(perPage) > 42 ? 42 : currentPage * Number(perPage);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {from} - {to} of {totalItems})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={event => {
+              setPerPage(Number(event.target.value));
+              setCurrentPage(1);
+            }}
+          >
             <option value="3">3</option>
-            <option value="5">5</option>
+            <option value="5" selected>
```

Comment:
The `selected` attribute on the option element is not the React way to set a default value for a select element. Instead, you should set the `value` attribute on the select element to match the value of the `perPage` state.